inbox/mailsync/backends/imap/generic.py

```diff
@@ -118,8 +119,10 @@ def __init__(self, account_id, namespace_id, folder_name,
 
         with session_scope(namespace_id) as db_session:
             try:
+                folder_stored_name = folder_name\
+                    .rstrip()[:MAX_FOLDER_NAME_LENGTH]
                 folder = db_session.query(Folder). \
-                    filter(Folder.name == folder_name,
+                    filter(Folder.name == folder_stored_name,
```

Comment:
Hmmmm or at least put it as a method on the Folder model or something like that. Generally in Python you can override comparator behaviour by defining a __cmp__ method on the object, but I'm not sure how nice that will play with SQLAlchemy's declarative declarations here.